### PR TITLE
[libmysql] Fix could not find package config files

### LINKED ISSUE #22057: [poco：mysql] Build error
**Host Environment**
- OS: Windows
- Compiler: VS 2022

**To Reproduce**
./vcpkg install poco[mysql]:x64-windows-static

**Failure logs** 
-Cut and paste the appropriate build messages from the console output.
-Please attach any additional failure logs mentioned in the console output.

**Additional context**
-- Using source at D:/dev/vcpkg/buildtrees/poco/src/f788bc88ea-d4759b3a2b.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows-static
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: ninja -v
    Working Directory: D:/dev/vcpkg/buildtrees/poco/x64-windows-static-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\dev\vcpkg\buildtrees\poco\config-x64-windows-static-out.log
[Uploading config-x64-windows-static-out.log…]()

### PR DESCRIPTION
**Pull request by LilyWangLL**
- #### What does your PR fix?  
  Fixes #22057

Fix ports which depend on libmysql build error:
```
Could not find a package configuration file provided by
  "unofficial-libmysql" with any of the following names:

    unofficial-libmysqlConfig.cmake
    unofficial-libmysql-config.cmake
```

### CODE REVIEW

File: ports/libmysql/portfile.cmake [hunk 1]
```diff
@@ -101,14 +99,14 @@ endif()
...
+file(RENAME "${CURRENT_PACKAGES_DIR}/debug/share" "${CURRENT_PACKAGES_DIR}/debug/${PORT}")
 file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/share")
 file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/debug/share")
-file(RENAME "${CURRENT_PACKAGES_DIR}/libmysql" "${CURRENT_PACKAGES_DIR}/share/libmysql")
-file(RENAME "${CURRENT_PACKAGES_DIR}/debug/libmysql" "${CURRENT_PACKAGES_DIR}/debug/share/libmysql")
+file(RENAME "${CURRENT_PACKAGES_DIR}/${PORT}" "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+file(RENAME "${CURRENT_PACKAGES_DIR}/debug/${PORT}" "${CURRENT_PACKAGES_DIR}/debug/share/${PORT}")
 
-vcpkg_cmake_config_fixup(CONFIG_PATH share/libmysql/unofficial-libmysql)
+vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-libmysql CONFIG_PATH share/libmysql/unofficial-libmysql)
```
Comment (dg0yt) [inline ports/libmysql/portfile.cmake:109]: Given the previous changes, you also need to replace `share/libmysql` with `share/${PORT}` here.

File: ports/libmysql/portfile.cmake [hunk 2]
```diff
@@ -101,14 +99,14 @@ endif()
...
-file(RENAME "${CURRENT_PACKAGES_DIR}/share" "${CURRENT_PACKAGES_DIR}/libmysql")
-file(RENAME "${CURRENT_PACKAGES_DIR}/debug/share" "${CURRENT_PACKAGES_DIR}/debug/libmysql")
+file(RENAME "${CURRENT_PACKAGES_DIR}/share" "${CURRENT_PACKAGES_DIR}/${PORT}")
+file(RENAME "${CURRENT_PACKAGES_DIR}/debug/share" "${CURRENT_PACKAGES_DIR}/debug/${PORT}")
 file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/share")
 file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/debug/share")
-file(RENAME "${CURRENT_PACKAGES_DIR}/libmysql" "${CURRENT_PACKAGES_DIR}/share/libmysql")
-file(RENAME "${CURRENT_PACKAGES_DIR}/debug/libmysql" "${CURRENT_PACKAGES_DIR}/debug/share/libmysql")
+file(RENAME "${CURRENT_PACKAGES_DIR}/${PORT}" "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+file(RENAME "${CURRENT_PACKAGES_DIR}/debug/${PORT}" "${CURRENT_PACKAGES_DIR}/debug/share/${PORT}")
```
Comment (dg0yt) [inline ports/libmysql/portfile.cmake:107]: Does libmysql actually find all the files in `share/${PORT}`, given that the rename is done after build?

**Review by dg0yt — COMMENTED** [review]
If possible, please try to make this port work also with single-config build, i.e. with absence of `debug` paths.

**Comment by cloudsnake** [comment]
thanks!
libmysql is ok!
but 
./vcpkg install poco[mysql]:x64-windows-static 
no error,no PocoDataMysqlmt.lib file!

File: ports/libmysql/portfile.cmake (hunk 2 above)
Comment (LilyWangLL) [reply ports/libmysql/portfile.cmake:107]: I just move config files and target files to `unofficial-libmysql`, other files in `share/${PORT}` keep the original structure.

File: ports/poco/portfile.cmake [hunk 3]
```diff
@@ -84,20 +83,20 @@ vcpkg_copy_tools(TOOL_NAMES cpspc f2cpsp PocoDoc tec arc AUTO_CLEAN)
 
 # Copy additional include files not part of any libraries
 if(EXISTS "${CURRENT_PACKAGES_DIR}/include/Poco/SQL")
-    file(COPY ${SOURCE_PATH}/Data/include DESTINATION ${CURRENT_PACKAGES_DIR})
+    file(COPY "${SOURCE_PATH}/Data/include" DESTINATION ${CURRENT_PACKAGES_DIR})
```
Comment (JackBoosY) [inline ports/poco/portfile.cmake:86]:
```suggestion
    file(COPY "${SOURCE_PATH}/Data/include" DESTINATION "${CURRENT_PACKAGES_DIR}")
```
etc.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
> thanks! libmysql is ok! but ./vcpkg install poco[mysql]:x64-windows-static no error,no PocoDataMysqlmt.lib file!

After new change, PocoDataMysqlmt.lib can install correct.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**